src/paint.js

```diff
@@ -108,25 +108,16 @@ function createDataArray(hypercubeData, layout) {
   return dataObject;
 }
 
-export default function paint($element, layout) {
+export default function paint($element, layout, component) {
+
   //set hypercube variable and call function on hcData to return data in a json format
   var hc = layout.qHyperCube,
     hcData = createDataArray(hc, layout);
 
   //create variables for number of bars allowed and the size of the dimension area for text
-  var numOfBarsAllowed,
-    dimWidth = Number(layout.props.section1.dimWidth);
-  //check that not too many bars are trying to be displayed
-  if (Math.floor($element.height() / 75) === 0) {
-    numOfBarsAllowed = 1;
-  } else {
-    numOfBarsAllowed = Math.floor($element.height() / 75);
-  }
+  var dimWidth = Number(layout.props.section1.dimWidth),
+    barsNum = layout.props.section2.barNum;
```

Comment:
there is no `barNum` only `barSize` for `section2`. Could you look into it if this affects something else as well?

Reply:
there is a `barNum` property in section 2!
code snippet from index.js
`
var barNum = {
  ref: 'props.section2.barNum',
  label: 'Quantity of bars showed(max. 50)',
  type: 'integer',
  defaultValue: 4,
  min: 1,
  max: 50
};
`

Reply:
Okay, could it be that I had an older version than? Do you know if barSize is an old property then?

Reply:
Yes
barNum was introduced in this PR
but barSize was there all along

Reply:
Okay, however `barNum` is not always present in `section2`. If you are looking at the sample app, then check out the other tabs

Reply:
It's a new property in definition, so old apps won't have that and will break. It's a non-issue as it's not been published yet.